src/tests/unit/tests/injected/adapters/scan-results-to-unified-results.test.ts

```diff
@@ -19,57 +20,154 @@ describe('ScanResults to Unified Results Test', () => {
+                m({
+                    id: 'id1',
+                    nodeResult: node1,
+                }),
+            )
+            .returns(() => fixResolutionStub);
+        fixResolutionCreatorMock
+            .setup(m =>
+                m({
+                    id: 'id1',
+                    nodeResult: node2,
+                }),
+            )
+            .returns(() => fixResolutionStub);
+        fixResolutionCreatorMock
+            .setup(m =>
+                m({
+                    id: 'id2',
+                    nodeResult: node3,
+                }),
+            )
+            .returns(() => fixResolutionStub);
+        const testSubject = new ConvertScanResultsToUnifiedResults(
+            generateGuidMock.object,
+            fixResolutionCreatorMock.object,
+            checkResolutionCreatorMock.object,
+        );
         const scanResultsStub: ScanResults = createTestResults();
-        expect(
-            convertScanResultsToUnifiedResults(scanResultsStub, generateGuidMock.object),
-        ).toMatchSnapshot();
+        expect(testSubject.automatedChecksConversion(scanResultsStub)).toMatchSnapshot();
         generateGuidMock.verifyAll();
     });
 
     test.each(nullIdentifiers)(
-        'convertScanResultsToNeedsReviewUnifiedResults provides a defined UnifiedResult instance %s',
+        'needsReviewConversion provides a defined UnifiedResult instance %s',
         scanResultStub => {
-            const unifiedResults = convertScanResultsToNeedsReviewUnifiedResults(
-                scanResultStub as ScanResults,
+            const testSubject = new ConvertScanResultsToUnifiedResults(
                 generateGuidMock.object,
+                fixResolutionCreatorMock.object,
+                checkResolutionCreatorMock.object,
             );
+            const unifiedResults = testSubject.needsReviewConversion(scanResultStub as ScanResults);
             expect(unifiedResults).toBeDefined();
         },
     );
 
-    test('needs review conversion works fine when there is no data in scanresults', () => {
+    test('needsReviewConversion works fine when there is no data in scanresults', () => {
+        const testSubject = new ConvertScanResultsToUnifiedResults(
+            generateGuidMock.object,
+            fixResolutionCreatorMock.object,
+            checkResolutionCreatorMock.object,
+        );
         const scanResultsStub: ScanResults = createTestResultsWithNoData();
-        expect(
-            convertScanResultsToNeedsReviewUnifiedResults(scanResultsStub, generateGuidMock.object),
-        ).toMatchSnapshot();
+        expect(testSubject.needsReviewConversion(scanResultsStub)).toMatchSnapshot();
     });
 
-    test('needs review conversion works with filled up passes, failures and incomplete values in scan results', () => {
+    test('needsReviewConversion works with filled up passes, failures and incomplete values in scan results', () => {
+        checkResolutionCreatorMock
```

Comment:
You can probably extract this out to a function that can be called with the right params on a single line. Would simplify it/make it easier to parse too.

Reply:
You can do the same for the fix portion as well. Since these mocks could share the same type, you can actually have one function which would take the mock too.

Reply:
Done!